packages/hydrogen/src/cart/cart-handler.ts

```diff
@@ -0,0 +1,158 @@
+import {Storefront} from '../storefront';
+import {getFormInput, CartActionInput} from './CartForm';
+import {parse as parseCookie} from 'worktop/cookie';
+import {cartGetDefault} from './queries/cartGetDefault';
+import {cartCreateDefault} from './queries/cartCreateDefault';
+import {cartLinesAddDefault} from './queries/cartLinesAddDefault';
+import {cartLinesUpdateDefault} from './queries/cartLinesUpdateDefault';
+import {cartLinesRemoveDefault} from './queries/cartLinesRemoveDefault';
+import {cartDiscountCodesUpdateDefault} from './queries/cartDiscountCodesUpdateDefault';
+import {cartBuyerIdentityUpdateDefault} from './queries/cartBuyerIdentityUpdateDefault';
+import {cartNoteUpdateDefault} from './queries/cartNoteUpdateDefault';
+import {cartSelectedDeliveryOptionsUpdateDefault} from './queries/cartSelectedDeliveryOptionsUpdateDefault';
+import {cartAttributesUpdateDefault} from './queries/cartAttributesUpdateDefault';
+import {cartMetafieldsSetDefault} from './queries/cartMetafieldsSetDefault';
+import {cartMetafieldDeleteDefault} from './queries/cartMetafieldDeleteDefault';
+
+export const cartGetIdDefault = (requestHeaders: Headers) => {
+  return () => {
+    const cookies = parseCookie(requestHeaders.get('Cookie') || '');
+    return cookies.cart ? `gid://shopify/Cart/${cookies.cart}` : undefined;
+  };
+};
+
+export const cartSetIdDefault = () => {
+  return (cartId: string, headers: Headers) => {
+    headers.append('Set-Cookie', `cart=${cartId.split('/').pop()}`);
```

Comment:
This probably should be minimally configurable. For example, when should the cart expire? At the moment it expires the moment the browser is closed, see this issue: https://github.com/Shopify/hydrogen/issues/919